src/network/connection_dispatcher_task.cpp

```diff
@@ -19,10 +19,10 @@ ConnectionDispatcherTask::ConnectionDispatcherTask(int num_handlers, int listen_
 
   // create worker threads.
   for (int task_id = 0; task_id < num_handlers; task_id++) {
-    auto handler = std::make_shared<ConnectionHandlerTask>(task_id, connection_handle_factory);
+    auto handler = new ConnectionHandlerTask(task_id, connection_handle_factory);
```

Comment:
Who owns this pointer from now on then?

Reply:
The `ConnectionDispatcherTask` owns the handler tasks, and is also responsible for [cleaning them up](https://github.com/cmu-db/terrier/pull/370/files/f5acda921b9dc1697e403291b819608cccc21acd#diff-cc4757df8e709f5364acdaf54c476669R40). I didn't write the code to handle when `ConnectionDispatcherTask` loses a thread from the thread registry, but the registry doesn't autonomously remove threads yet so thats okay for now.